x-pack/solutions/observability/plugins/apm/public/components/app/error_group_overview/error_group_list/use_error_group_list_data.tsx

```diff
@@ -34,10 +35,10 @@ const INITIAL_STATE_DETAILED_STATISTICS: DetailedStatistics = {
 };
 
 export function useErrorGroupListData({
-  renderedItems,
+  renderedItemIndices,
   sorting,
 }: {
-  renderedItems: ErrorGroupItem[];
+  renderedItemIndices: VisibleItemsStartEnd;
```

Comment:
out of interest, why use a tuple here, and not just renderedItemsStartIndex:number, renderedItemsEndIndex:number?

Reply:
A tuple makes sense because these values exist only as a pair. A start value and an end value, and the form is a stable `[start, end]` form. It maps to how the table is showing what items are visible, and the two values update together, never separately. As such, it doesn't make sense to split it out into separate, independent values.

Tuples are useful and really handy, plus TS has good support for them so why not make use of them?

Reply:
A tuple is much more likely to trash your hook dependencies, and the readibility is zero. It seems like an odd choice, but soit

Reply:
If we weren't memoizing the tuple creation, then yes, it would cause problems, but I'm not sure I agree about readability. I am much more used to using tuples in other languages, however. Not having invalid states is more important, since that was what was causing the request misfiring (the renderedItems array was causing the hook dependencies to be trashed in the first place). I'm not sure other approaches here would have been better without being unnecessarily verbose or falling back to using an object (which itself would also need careful handling to avoid trashing hook dependencies as well).

Reply:
> I am much more used to using tuples in other languages

That explains :)